src/boco_core/boco_window/window.odin

```diff
@@ -7,46 +7,137 @@ import sdl "vendor:sdl2"
+                #partial switch event.window.event {
+                    case .CLOSE:
+                        window_event.state = Window_state.Quit
+                        is_ready_to_close = true
+                        log.info("Closing Window:", name)
+                    case .FOCUS_GAINED:
+                        window_event.state = Window_state.Focus
+                        log.info("Focusing Window:", name)
+                    case .FOCUS_LOST:
+                        window_event.state = Window_state.Unfocus
+                    case .MOVED:
+                        window_event.state = Window_state.Move
+                    case .RESIZED:
+                        window_event.state = Window_state.Resize
+                    case .SIZE_CHANGED:
+                        window_event.state = Window_state.Resize
+                    case .MINIMIZED:
+                        window_event.state = Window_state.Minimize
+                    case .MAXIMIZED:
+                        window_event.state = Window_state.Maximize
+                    case .RESTORED:
+                        window_event.state = Window_state.Restore
+                    case .ENTER:
+                        window_event.state = Window_state.Enter
+                    case .LEAVE:
+                        window_event.state = Window_state.Leave
+                }   
+            }
+        case .KEYDOWN:
+            window_event.state = Window_state.Key
+            key_from_code := key_name_from_code
+            window_event.key_event.key.name = key_from_code[event.key.keysym.scancode]
+            window_event.key_event.key.code = cast(u32)event.key.keysym.scancode
+            window_event.key_event.state = Key_state.Pressed
+            log.info("_____________________")
+            log.info("Key Pressed on window:", name)
+            log.info(window_event.key_event.key.name)
+            log.info("_____________________")
+            receive_event(parent_window, &window_event)
+        case .KEYUP:
+            window_event.state = Window_state.Key
+            key_from_code := key_name_from_code
+            window_event.key_event.key.name = key_from_code[event.key.keysym.scancode]
+            window_event.key_event.key.code = cast(u32)event.key.keysym.scancode
+            window_event.key_event.state = Key_state.Released
+            log.info("_____________________")
+            log.info("Key Released on window:", name)
+            log.info(window_event.key_event.key.name)
+            log.info("_____________________")
+            receive_event(parent_window, &window_event)
         }
     }
-    update_child_windows(window)
-    
-    return true
 }
 
-create_child_window :: proc(using window: ^Window){
+create_child_window :: proc(using window: ^Window, title: cstring = "BOCO"){
     new_window:Window
+    new_window.parent_window = window
     append(&child_windows, new_window)
+    init(&child_windows[len(child_windows) - 1], title)
```

Comment:
Can tidy up by doing something like
```
append(&child_windows, Window{})
new_window := &child_windows[len(child_windows) - 1]
new_window.parent_window = window
init(new_window)
```
but not too important, just that init line is kind of nasty